.github/workflows/net-perf-gke.yaml

```
@@ -0,0 +1,252 @@
+        uses: google-github-actions/auth@67e9c72af6e0492df856527b474995862b7b6591 # v2.0.0
+        with:
+          credentials_json: '${{ secrets.GCP_PR_SA_KEY }}'
+
+      - name: Set up gcloud CLI
+        uses: google-github-actions/setup-gcloud@825196879a077b7efa50db2e88409f44de4635c2 # v2.0.0
+        with:
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          version: "405.0.0"
+
+      - name: Install gke-gcloud-auth-plugin
+        run: |
+          gcloud components install gke-gcloud-auth-plugin
+
+      - name: Display gcloud CLI info
+        run: |
+          gcloud info
+
+      - name: Create GKE cluster
+        run: |
+          gcloud container clusters create ${{ env.clusterName }}-${{ matrix.index }} \
+            --labels "usage=${{ github.repository_owner }}-${{ github.event.repository.name }},owner=${{ steps.vars.outputs.owner }}" \
+            --zone ${{ env.gcp_zone }} \
+            --cluster-version ${{ env.k8s_version }} \
+            --enable-ip-alias \
+            --create-subnetwork="range=/26" \
+            --cluster-ipv4-cidr="/21" \
+            --services-ipv4-cidr="/24" \
+            --image-type COS_CONTAINERD \
+            --num-nodes 2 \
+            --machine-type n2-standard-2 \
+            --disk-type pd-standard \
+            --disk-size 20GB \
+            --node-taints ignore-taint.cluster-autoscaler.kubernetes.io/cilium-agent-not-ready=true:NoExecute
+
+      - name: Get cluster credentials
+        run: |
+          gcloud container clusters get-credentials ${{ env.clusterName }}-${{ matrix.index }} --zone ${{ env.gcp_zone }}
+
+      - name: Wait for images to be available
+        timeout-minutes: 30
+        shell: bash
+        run: |
+          for image in cilium-ci operator-generic-ci hubble-relay-ci ; do
+            until docker manifest inspect quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/$image:${{ steps.vars.outputs.sha }} &> /dev/null; do sleep 45s; done
+          done
+
+      - name: Create custom IPsec secret
+        if: ${{ matrix.encryption == 'ipsec' }}
+        run: |
+          kubectl create -n kube-system secret generic cilium-ipsec-keys --from-literal=keys="15 rfc4106(gcm(aes)) $(echo $(dd if=/dev/urandom count=20 bs=1 2> /dev/null | xxd -p -c 64)) 128"
+
+      - name: Install Cilium
+        id: install-cilium
+        run: |
+          cilium install --dry-run-helm-values ${{ steps.vars.outputs.cilium_install_defaults }}
+          cilium install ${{ steps.vars.outputs.cilium_install_defaults }}
+
+      - name: Wait for Cilium to be ready
+        run: |
+          cilium status --wait --wait-duration=10m
+          kubectl get pods -n kube-system
+          kubectl -n kube-system exec daemonset/cilium -- cilium-dbg status
+
+      - name: Run perf test (${{ matrix.name }})
+        id: run-perf
+        run: |
+          mkdir output
+          cilium connectivity perf --duration=30s --host-net=true --pod-net=true --report-dir=./output
```

I'd recommend bumping this duration up a bit, which will help tone down the variance. Three minutes would be ideal, but we should at least do one minute:

```suggestion
          cilium connectivity perf --duration=60s --host-net=true --pod-net=true --report-dir=./output
```

---

Talking offline, we'll keep it at 30s and increase as needed. The goal is to try and balance this duration value and the duration of the workflow overall.